src/MembershipContextFactory.php

```diff
@@ -4,97 +4,73 @@
-use WMDE\Fundraising\Store\Installer;
 
 /**
  * @licence GNU GPL v2+
  * @author Jeroen De Dauw < jeroendedauw@gmail.com >
  */
-class MembershipContextFactory implements ServiceProviderInterface {
-
-	private $config;
+class MembershipContextFactory {
 
 	/**
-	 * @var Container
+	 * Use this constant for MappingDriverChain::addDriver
 	 */
-	private $pimple;
+	public const DOCTRINE_ADDITIONAL_ENTITIES = [
+		'WMDE\Fundraising\MembershipContext\DataAccess\DoctrineEntities' => __DIR__ . '/DoctrineEntities'
+	];
 
-	private $addDoctrineSubscribers = true;
+	public const ENTITY_PATHS = [
+		__DIR__ . '/DataAccess/DoctrineEntities/'
+	];
 
-	public function __construct( array $config ) {
-		$this->config = $config;
-		$this->pimple = $this->newPimple();
-	}
+	private $addDoctrineSubscribers = true;
 
-	private function newPimple(): Container {
-		$pimple = new Container();
-		$this->register( $pimple );
-		return $pimple;
-	}
+	private array $config;
+	private Configuration $doctrineConfig;
+	private ?RandomMembershipTokenGenerator $tokenGenerator;
 
-	public function register( Container $container ): void {
-		$container['dbal_connection'] = function() {
-			return DriverManager::getConnection( $this->config['db'] );
-		};
-
-		$container['entity_manager'] = function() {
-			$entityManager = ( new StoreFactory( $this->getConnection(), $this->getVarPath() . '/doctrine_proxies' ) )
-				->getEntityManager();
-			if ( $this->addDoctrineSubscribers ) {
-				$entityManager->getEventManager()->addEventSubscriber(
-					$this->newDoctrineMembershipPrePersistSubscriber()
-				);
-			}
-
-			return $entityManager;
-		};
-
-		$container['fundraising.membership.application.token_generator'] = function() {
-			return new RandomMembershipTokenGenerator(
-				$this->config['token-length'],
-				new \DateInterval( $this->config['token-validity-timestamp'] )
-			);
-		};
-
-		$container['fundraising.membership.application.authorizer.class'] = DoctrineApplicationAuthorizer::class;
-		// @todo Consider if the tokens should not better be method parameters (see ApplicationAuthorizer interface)
-		$container['fundraising.membership.application.authorizer.update_token'] = null;
-		$container['fundraising.membership.application.authorizer.access_token'] = null;
-
-		$container['fundraising.membership.application.authorizer'] = $container->factory( function ( Container $container ): ApplicationAuthorizer {
-			return new $container['fundraising.membership.application.authorizer.class'](
-				$container['entity_manager'],
-				$container['fundraising.membership.application.authorizer.update_token'],
-				$container['fundraising.membership.application.authorizer.access_token']
-			);
-		} );
-	}
-
-	public function getConnection(): Connection {
-		return $this->pimple['dbal_connection'];
+	public function __construct( array $config, Configuration $doctrineConfig, ?RandomMembershipTokenGenerator $tokenGenerator = null ) {
```

Comment:
Please use the `MembershipTokenGenerator` instead of the concrete implementation, otherwise setting different implementations -  the purpose of this param - won't work